src/redux/consentsSlice.js

```diff
@@ -0,0 +1,66 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getConsents } from 'api';
+
+const initialState = {
+  definitions: [],
+  labels: [],
+  checked: [],
+  loading: false,
+  error: ''
+};
+
+export const fetchConsents = createAsyncThunk(
+  'consents',
+  async (publisherId, { rejectWithValue }) => {
+    try {
+      const responseData = await getConsents(publisherId);
+      return responseData;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  }
+);
+
+export const consentsSlice = createSlice({
+  name: 'consents',
+  initialState,
+  reducers: {
+    setChecked(state, { payload }) {
+      state.checked[payload] = !state.checked[payload];
+    }
+  },
+  extraReducers: {
+    [fetchConsents.pending]: state => {
+      state.loading = true;
+    },
+    [fetchConsents.fulfilled]: (state, { payload }) => {
+      if (payload.responseData && payload.responseData.consents) {
+        const definitions = payload.responseData.consents.map(element => {
+          return {
+            name: element.name,
+            version: element.version,
+            required: element.required
+          };
+        });
+        const labels = payload.responseData.consents.map(
+          element => element.label
+        );
+        const checked = new Array(definitions.length).fill(false);
+        state.definitions = definitions;
+        state.labels = labels;
+        state.checked = checked;
```

Comment:
I'm wondering if we need a separate array of labels states. We could simply keep the whole object from API (publisher contents resource as a single source of true) and add a 'checked' state to track the temporary state. So it would be (sample): 
```
const initialState = {
  consents: [
  {
     "broadcasterId": 0,
      "name": "terms",
       "version": "1",
       "value": "https://cleeng.com/cleeng-user-agreement",
       "label": "I accept the <a href=\"https://cleeng.com/cleeng-user-agreement\" target=\"_blank\">Terms and Conditions</a> of Cleeng.",
        "required": true
  }],
  checked: [false],
  loading: false,
  error: ''
};
```
The usage of 'labels' in Consents component could be changed from `labels[index]` to simply `label`